File: ex_obj.js
'use strict';
//Watch Guard 工具箱
const WG = require('./wg_utils');
const cfg = require('./wg_config');
var wg_utils = new WG(cfg.repos_data_host, cfg.repos_data_path);
//參數 input_sdate_str 始開時間
process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
var input_sdate_Str = '2017-12-14 06:00:00 GMT';
if (process.argv.length >= 3) {
    input_sdate_Str = process.argv[2];
}
wg_utils.Auth_Login(cfg.Auth_Login_Host, cfg.Auth_Login_Path_, cfg.user_pwd);

var index_ = 0;
var link_paths = [
    '/',
    '/control/device',
    '/control/devices',
    '/log/log_groups'
];

let app =function (){}
app.vIntervalTimer = null;
app.GetWGLog = function () {
    console.error(index_);
    if (index_ >= link_paths.length) {
        clearInterval(app.vIntervalTimer);
        return;
    }
    var path_ = link_paths[index_];
    index_++;
    console.log(index_, path_);
    wg_utils.HttpGet(cfg.Auth_Login_Host, path_);
}
app.run = function () {
    app.vIntervalTimer = setInterval(app.GetWGLog, 2000);
}
app.run();

File: ex_obj.py



#!/usr/bin/env python

# import modules used here -- sys is a very standard one
import sys
import wg_utils
import wg_config
import http.client
import urllib
import ssl
import time

# Gather our code in a main() function
def main():
    print(sys.argv)
    host=wg_config.Auth_Login_Host()
    path="/auth/login"
    params = urllib.parse.urlencode({'username': wg_config.WG_Username(), 'password': wg_config.WG_Password(), 'from_page': '/'})
    wg_u=wg_utils.WG(host);
    wg_u.Auth_Login(path,params)
    link_paths = [
  '/',
  '/control/device',
  '/control/devices',
  '/log/log_groups',
  '/log/log_vpns',
  '/log/log_devices?_search=false&rows=10000&page=1&sidx=name&sord=asc',
  '/log/log_grid_columns?l_t=servers',
    ];
    for link_ in link_paths :
        wg_u.HttpGet( link_)
        time.sleep(3)
# Standard boilerplate to call the main() function to begin
# the program.
if __name__ == '__main__':
    main()
